CsCore/PlainNetClassLib/src/Plugins/CsCore/com/csutil/http/apis/OpenAi.cs

```diff
@@ -38,6 +39,101 @@ public Task<Image.Response> TextToImage(Image.Request requestParams)
             return request.WithAuthorization(apiKey).WithJsonContent(requestParams).GetResult<Image.Response>();
         }
 
+        /// <summary> https://beta.openai.com/docs/api-reference/images/create </summary>
+        public Task<Vision.Response> ImageToText(Vision.Request requestParams, string imageURL)
+        {
+            var request = new Uri("https://api.openai.com/v1/chat/completions").SendPOST();
```

Comment:
I think the chat/completions endpoint is already used in a different place in the OpenAi.cs class, so I think you can just modify that logic and add that it optionally also supports to add an image to it

Reply:
This might also allow you to get rid of your entire Vision class and only add new parameter(s) to the existing request and response classes

Reply:
Use the Gpt.Response class 14dc66e24fe5c5752cdc32e1d96742a5cf9e2d6b

Reply:
extend the request class to a JsonRequest where we add the response_format field

Reply:
d95ac26e0e715d048433b5400cb7f25766840da3